lib/tapioca/dsl/compilers/job_iteration.rb

```diff
@@ -0,0 +1,96 @@
+# typed: strict
+# frozen_string_literal: true
+
+return unless defined?(JobIteration::Iteration)
+
+require "prism"
+
+module Tapioca
+  module Dsl
+    module Compilers
+      class JobIteration < Compiler
+        extend T::Sig
+
+        ConstantType = type_member { { fixed: T.class_of(::JobIteration::Iteration) } }
+
+        sig { override.void }
+        def decorate
+          return unless constant.instance_methods(false).include?(:build_enumerator)
+
+          root.create_path(constant) do |job|
+            method = constant.instance_method(:build_enumerator)
+            constant_name = name_of(constant)
+            parameters = compile_method_parameters_to_rbi(method).reject do |typed_param|
+              typed_param.param.name == "cursor"
+            end
+
+            expanded_parameters = parameters.flat_map do |typed_param|
+              hash_param = typed_param.type.match(/\A\{.*\}\z/)
+              if hash_param
+                key_value_pairs = parse_hash_parameter(typed_param)
+                key_value_pairs.map do |key, value|
+                  create_kw_param(key, type: value)
+                end
+              else
+                typed_param
+              end
+            end
+
+            return_type = compile_method_return_type_to_rbi(method)
+
+            job.create_method(
+              "perform_later",
+              parameters: expanded_parameters,
+              return_type: "T.any(#{constant_name}, FalseClass)",
+              class_method: true,
+            )
+
+            job.create_method(
+              "perform_now",
+              parameters: expanded_parameters,
+              return_type: return_type,
```

Comment:
The return type of `build_enumerator` will generally be an enumerator of some kind. The return type of` perform_now` is `void` for job-iteration jobs: https://github.com/Shopify/job-iteration/blob/7c808419d82860baf3aefaff9b32e08e5c415afb/lib/job-iteration/iteration.rb#L113